cmake/KokkosConfig.cmake.in

```diff
@@ -19,7 +19,17 @@ include("${Kokkos_CMAKE_DIR}/KokkosTargets.cmake")
 include("${Kokkos_CMAKE_DIR}/KokkosConfigCommon.cmake")
 unset(Kokkos_CMAKE_DIR)
 
-# check for conflicts
+# components deprecation
+if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS)
+  message(WARNING "'launch_compiler' is deprecated and will be removed in the future. Use the 'Kokkos_FORCE_COMPILER_LAUNCHER' option instead.")
+  set(Kokkos_GLOBAL_COMPILATION true)
```

Comment:
Nitpick: I'd slightly prefer ON/OFF
(not blocking though)

Reply:
done in https://github.com/kokkos/kokkos/pull/7581/commits/35431f5b6c0c0c3d7c3ad53db1b8c7112755ea0d